vnext/Microsoft.ReactNative/Modules/Animated/InterpolationAnimatedNode.cpp

```diff
@@ -24,49 +24,66 @@ InterpolationAnimatedNode::InterpolationAnimatedNode(
-  m_propertySet.StopAnimation(s_valueName);
-  m_rawValueAnimation = nullptr;
-  m_offsetAnimation = nullptr;
+
+  if (m_useComposition) {
+    m_propertySet.StopAnimation(s_valueName);
+    m_propertySet.StopAnimation(s_valueName);
+    m_rawValueAnimation = nullptr;
+    m_offsetAnimation = nullptr;
+  }
 }
 
 void InterpolationAnimatedNode::OnAttachToNode(int64_t animatedNodeTag) {
+  assert(HasCompatibleAnimationDriver(animatedNodeTag));
   assert(m_parentTag == s_parentTagUnset);
   m_parentTag = animatedNodeTag;
 
-  const auto [rawValueAnimation, offsetAnimation] = [this]() {
-    if (const auto manager = m_manager.lock()) {
-      if (const auto parent = manager->GetValueAnimatedNode(m_parentTag)) {
-        const auto compositor = Microsoft::ReactNative::GetCompositor();
-
-        const auto rawValueAnimation = CreateExpressionAnimation(compositor, *parent);
-        rawValueAnimation.Expression(
-            GetExpression(s_parentPropsName + static_cast<winrt::hstring>(L".") + s_valueName));
-
-        const auto offsetAnimation = CreateExpressionAnimation(compositor, *parent);
-        offsetAnimation.Expression(
-            L"(" +
-            GetExpression(
-                s_parentPropsName + static_cast<winrt::hstring>(L".") + s_offsetName + L" + " + s_parentPropsName +
-                L"." + s_valueName) +
-            L") - this.target." + s_valueName);
-
-        return std::make_tuple(rawValueAnimation, offsetAnimation);
+  if (m_useComposition) {
+    const auto [rawValueAnimation, offsetAnimation] = [this]() {
+      if (const auto manager = m_manager.lock()) {
+        if (const auto parent = manager->GetValueAnimatedNode(m_parentTag)) {
+          const auto compositor = Microsoft::ReactNative::GetCompositor();
+
+          const auto rawValueAnimation = CreateExpressionAnimation(compositor, *parent);
+          rawValueAnimation.Expression(
+              GetExpression(s_parentPropsName + static_cast<winrt::hstring>(L".") + s_valueName));
+
+          const auto offsetAnimation = CreateExpressionAnimation(compositor, *parent);
+          offsetAnimation.Expression(
+              L"(" +
+              GetExpression(
+                  s_parentPropsName + static_cast<winrt::hstring>(L".") + s_offsetName + L" + " + s_parentPropsName +
+                  L"." + s_valueName) +
+              L") - this.target." + s_valueName);
+
+          return std::make_tuple(rawValueAnimation, offsetAnimation);
+        }
       }
-    }
-    return std::tuple<comp::ExpressionAnimation, comp::ExpressionAnimation>(nullptr, nullptr);
-  }();
+      return std::tuple<comp::ExpressionAnimation, comp::ExpressionAnimation>(nullptr, nullptr);
+    }();
 
-  m_propertySet.StartAnimation(s_valueName, rawValueAnimation);
-  m_propertySet.StartAnimation(s_offsetName, offsetAnimation);
+    m_propertySet.StartAnimation(s_valueName, rawValueAnimation);
+    m_propertySet.StartAnimation(s_offsetName, offsetAnimation);
```

Comment:
Unrelated bug - this looks like another case where we're animating the offset rather than the value, though I think this is okay if we can't set an offset value to an interpolation node.